src/ccs/search_state.py

```diff
@@ -76,6 +93,8 @@ def activate_and(n, propagated_specificity):
         def activate_or(n, propagated_specificity):
             nonlocal or_specificities
             prev_spec = or_specificities.get(n, Specificity(0, 0, 0, 0))
+            if propagated_specificity is None:
+                return prev_spec
```

Comment:
I don't love this, but I haven't fully decided what would be better

Reply:
Yeah it's a little gross, but not terrible. I'll take a look after the merge, I have a couple ideas.